pkg/task/cluster.go

```diff
@@ -32,6 +30,27 @@ func (ti *TaskInterface) ProcessClusters(directory string) error {
 		cluster.Name = DefaultName
 	}
 
+	// Prefer non-deprecated values over deprecated values
+	if cluster.Connection == nil {
+		cluster.Connection = &model.ClusterConnectionSpec{}
+	}
+	if cluster.Connection.Server == "" {
+		cluster.Connection.Server = cluster.URL
+		cluster.URL = ""
+	}
+	if cluster.Connection.CertificateAuthority == "" {
+		ca, err := base64.StdEncoding.DecodeString(cluster.CAData)
+		if err != nil {
+			return err
```

Comment:
I believe you might get an error if `cluster.CAData` is empty in this case. Might want to make sure that lines up with the previous behavior.

Reply:
Looks fine. Decoding empty `string` is just an empty `[]byte` and results in no `certificate-authority-data` in the kubectl config.